bugs/lp-1889555.txt

```diff
@@ -0,0 +1,1210 @@
+Fetched 429 kB in 1s (409 kB/s)
+Preconfiguring packages ...
+(Reading database ... 59604 files and directories currently installed.)
+Preparing to unpack .../cloud-init_20.3-2-g371b392c-0ubuntu1~20.04.1_all.deb ...
+Unpacking cloud-init (20.3-2-g371b392c-0ubuntu1~20.04.1) over (20.1-10-g71af48df-0ubuntu5) ...
+Setting up cloud-init (20.3-2-g371b392c-0ubuntu1~20.04.1) ...
+Installing new version of config file /etc/cloud/cloud.cfg.d/05_logging.cfg ...
+Installing new version of config file /etc/cloud/templates/chef_client.rb.tmpl ...
+Installing new version of config file /etc/cloud/templates/hosts.freebsd.tmpl ...
+Installing new version of config file /etc/cloud/templates/hosts.suse.tmpl ...
+Installing new version of config file /etc/cloud/templates/resolv.conf.tmpl ...
+Reconfiguring grub install device due to mismatch (LP: #1889555)
+   grub should use /dev/disk/by-id/nvme-Amazon_Elastic_Block_Store_vol0e030315c300ed9f8 but is configured for /dev/sda
+Processing triggers for rsyslog (8.2001.0-1ubuntu1) ...
+
+root@ip-172-31-46-171:/home/ubuntu# debconf-show grub-pc | grep "install_devices:"
+* grub-pc/install_devices: /dev/disk/by-id/nvme-Amazon_Elastic_Block_Store_vol0e030315c300ed9f8
+  grub-efi/install_devices:
+
+root@ip-172-31-46-171:/home/ubuntu# reboot
+root@ip-172-31-46-171:/home/ubuntu# Connection to 3.14.71.24 closed by remote host.
+Connection to 3.14.71.24 closed.
+(cloud-init) newt:cloud-init (croessner-master) $ ssh ubuntu@3.14.71.24
+Welcome to Ubuntu 20.04 LTS (GNU/Linux 5.4.0-1009-aws x86_64)
+# motd message
+
+(cloud-init) newt:cloud-init (croessner-master) $ aws ec2 terminate-instances --instance-ids i-08204591562ea8dd1
+
+
+# Bionic
+newt:manual (fix-nvme-grub*) $ aws ec2 describe-images --filters "Name=name,Values=Ubuntu 18.04*" | grep ImageId
+            "ImageId": "ami-0c858b55bf9119371",
+            "ImageId": "ami-0e7ef7e5358591bc2",
+newt:manual (fix-nvme-grub*) $ aws ec2 run-instances --image-id ami-0e7ef7e5358591bc2 --count 1 --instance-type t3a.medium --key-name newt | grep InstanceId
+            "InstanceId": "i-032ea99544b1d0649",
+newt:manual (fix-nvme-grub*) $ aws ec2 describe-instances --instance-ids i-032ea99544b1d0649 | grep PublicIpAddress
+                    "PublicIpAddress": "18.191.217.15",
+newt:manual (fix-nvme-grub*) $ ssh ubuntu@18.191.217.15
+Warning: Permanently added '18.191.217.15' (ECDSA) to the list of known hosts.
+# motd message
+
+ubuntu@ip-172-31-32-175:~$ cloud-init --version
+/usr/bin/cloud-init 19.2-24-ge7881d5c-0ubuntu1~18.04.1
+ubuntu@ip-172-31-32-175:~$ lsblk | grep nvme
+nvme0n1     259:0    0  100G  0 disk
+└─nvme0n1p1 259:1    0  100G  0 part /
+
+ubuntu@ip-172-31-32-175:~$ debconf-show grub-pc | grep "install_devices:"
+debconf: DbDriver "passwords" warning: could not open /var/cache/debconf/passwords.dat: Permission denied
+* grub-pc/install_devices: /dev/sda
```

Comment:
Is this significant that this bionic image doesn't have a grub-efi debconfig setting?

Reply:
I noticed that, but no, the issue only manifests on the grub-pc installs.